### ISSUE TITLE
Remove dates property from Model

### ISSUE THREAD

**Issue by halaei** [laravel/ideas#1676]
Model's `$dates` property was there before casts property was added. Now `$casts` property supports `date` and `datetime`, so $dates property is not required anymore. Other than that, the name of the property is a little bit confusing; is it going to cast a field to Date (as the start of each day) or DateTime (any second of a day)? I also confuse this field with `$timestamps` once in a while.
So I think of removing it.

```php

    /**
     * The attributes that should be cast to native types.
     *
     * @var array
     */
    protected $casts = [];

    /**
     * The attributes that should be mutated to dates.
     *
     * @var array
     */
    protected $dates = [];

```

Please let me know your opinion on this.
Thanks.

**Comment by Sladewill** [comment]
It should be deprecated with a warning as some people may still be using this whilst upgrading to the latest versions.

**Comment by donnysim** [comment]
I've personally never in my whole time using laravel, used the $dates field even once. Always the $casts, so I don't have anything against deprecating and removing it.

**Comment by martinbean** [comment]
> I've personally never in my whole time using Laravel

@donnysim Maybe so, but you’re also not the only person who uses Laravel. Plenty of other people do, and plenty will have used the `$dates` property. I have, for example.

**Comment by mfn** [comment]
I looked a bit around in the Eloquent model and HasAttributes trait. It truly seems that it would be possible to complete git rid of them without loosing functionality (sans the breaking change).

**Comment by rimace** [comment]
@martinbean I guess this is about consistency. If there is a `$dates` property, there should also be a `$floats` and a `$ints` property or handle every cast using the `$casts` property.
So I support this suggestion as it gets rid of code that solves something that is already solved with the `$casts` property.

**Comment by martinbean** [comment]
@rimace That’s a slippery slope, as I imagine they’ll be _loads_ of inconsistencies like that throughout the framework if you adopt that mindset.

Instead, the pragmatic upset or this particular instance is: there will be Laravel projects in the wild that use the `$dates` property in models, that this change will break.

**Comment by rimace** [comment]
@martinbean Yes, it will definitely break things, like the Framework does with every major and minor version. If you look at the upgrade guides there is a lot of stuff that was removed and broke applications that didn't adapt. That's why I think the argument "it will break things" is not really strong.
If we adopt that mindset a software - including Laravel - could never be were it is today, as breaking changes are part of development in general.
So instead on looking at the breaking application, I'd rather like to discuss if this is a good or bad practice.

**Comment by ManojKiranA** [comment]
@staudenmeir what do you think

**Comment by staudenmeir** [comment]
I'm torn on this one, but I'm leaning towards keeping `$dates`. It's good to remove superfluous properties, but I'm sure *a lot* of people use it. For such a huge breaking change, the benefit is rather small.

Is there a good way to deprecate properties? At least in PhpStorm, you don't get a notice when overriding a protected property.

**Comment by rimace** [comment]
@staudenmeir You could check for the `$dates` call in `__get()` and trigger an `E_USER_NOTICE` or log it. Currently the only way I find "soft".
